core/modules/qhttp/Response.cc

```diff
@@ -0,0 +1,182 @@
+    {102, "Processing"},                        {200, "OK"},
+    {201, "Created"},                           {202, "Accepted"},
+    {203, "Non-Authoritative Information"},     {204, "No Content"},
+    {205, "Reset Content"},                     {206, "Partial Content"},
+    {207, "Multi-Status"},                      {208, "Already Reported"},
+    {226, "IM Used"},                           {300, "Multiple Choices"},
+    {301, "Moved Permanently"},                 {302, "Found"},
+    {303, "See Other"},                         {304, "Not Modified"},
+    {305, "Use Proxy"},                         {307, "Temporary Redirect"},
+    {308, "Permanent Redirect"},                {400, "Bad Request"},
+    {401, "Unauthorized"},                      {402, "Payment Required"},
+    {403, "Forbidden"},                         {404, "Not Found"},
+    {405, "Method Not Allowed"},                {406, "Not Acceptable"},
+    {407, "Proxy Authentication Required"},     {408, "Request Timeout"},
+    {409, "Conflict"},                          {410, "Gone"},
+    {411, "Length Required"},                   {412, "Precondition Failed"},
+    {413, "Payload Too Large"},                 {414, "URI Too Long"},
+    {415, "Unsupported Media Type"},            {416, "Range Not Satisfiable"},
+    {417, "Expectation Failed"},                {421, "Misdirected Request"},
+    {422, "Unprocessable Entity"},              {423, "Locked"},
+    {424, "Failed Dependency"},                 {426, "Upgrade Required"},
+    {428, "Precondition Required"},             {429, "Too Many Requests"},
+    {431, "Request Header Fields Too Large"},   {500, "Internal Server Error"},
+    {501, "Not Implemented"},                   {502, "Bad Gateway"},
+    {503, "Service Unavailable"},               {504, "Gateway Timeout"},
+    {505, "HTTP Version Not Supported"},        {506, "Variant Also Negotiates"},
+    {507, "Insufficient Storage"},              {508, "Loop Detected"},
+    {510, "Not Extended"},                      {511, "Network Authentication Required"},
+};
+
+
+std::unordered_map<std::string, const std::string> contentTypesByExtension = {
+    {".css", "text/css"},
+    {".gif", "image/gif"},
+    {".htm", "text/html"},
+    {".html", "text/html"},
+    {".jpg", "image/jpeg"},
+    {".js", "application/javascript"},
+    {".png", "image/png"},
+};
+
+} // anon namespace
+
+
+namespace lsst {
+namespace qserv {
+namespace qhttp {
+
+Response::Response(
+    std::shared_ptr<ip::tcp::socket> socket,
+    DoneCallback const &doneCallback)
+:
+    _socket(socket),
+    _doneCallback(doneCallback)
+{
+}
+
+
+void Response::sendStatus(unsigned int status)
+{
+    this->status = status;
+    std::string statusStr = responseStringsByCode[status];
+    std::ostringstream entStr;
+    entStr << "<html>" << std::endl;
+    entStr << "<head><title>" << status << " " << statusStr << "</title></head>" << std::endl;
+    entStr << "<body bgcolor=\"white\">" << std::endl;
```

Comment:
There should not be any styling within the tags. All has to be done via CSS stylesheets.
My second comment is that the very first line of each HTML5-complient document must be:
<!DOCTYPE html>
<html lang="en">
...